### PR TITLE
Finish view profile and view all user

### PR DESCRIPTION
**Pull request by manhnhhedspi**

### CODE REVIEW

**Comment by phuongbk** [comment]
Chưa có Init model à?

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  def create
+    @user = User.new user_params
+    if @user.save
+      flash[:success] = "#{t("controllers.users.create.flash_success1")}
+                         #{@user.name}
+                         #{t("controllers.users.create.flash_success2")}"
+      redirect_to @user
+    else
+      render "new"
+    end
```
Comment (tunv) [inline app/controllers/users_controller.rb]: render :new

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+  def create
+    @user = User.new user_params
+    if @user.save
+      flash[:success] = "#{t("controllers.users.create.flash_success1")}
+                         #{@user.name}
+                         #{t("controllers.users.create.flash_success2")}"
+      redirect_to @user
+    else
+      render "new"
```
Comment (tunv) [inline app/controllers/users_controller.rb]: change to render :new

File: app/helpers/users_helper.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+module UsersHelper
+  def gravatar_for user
+    gravatar_id = Digest::MD5.hexdigest user.email.downcase
+    gravatar_url = "https://secure.gravatar.com/avatar/#{gravatar_id}"
```
Comment (tunv) [inline app/helpers/users_helper.rb]: add to rails_config

File: app/models/user.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+class User < ActiveRecord::Base
+  before_save{self.email = email.downcase}
+  validates :name, presence: true
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+  validates :email, presence: true, length: {maximum: 255},
+                    format: {with: VALID_EMAIL_REGEX},
+                    uniqueness: {case_sensitive: false}
+  has_secure_password
+  validates :password, presence: true, length: {minimum: 6}
```
Comment (tunv) [inline app/models/user.rb]: add 6 to rails config

File: app/views/users/new.html.erb [hunk 5]
```diff
@@ -0,0 +1,21 @@
+<h1><%= t ".signup" %></h1>
+<div class="row">
+  <div class="col-md-6 col-md-offset-3">
+    <%= form_for(@user) do |f| %>
+      <%= render 'shared/error_messages' %>
```
Comment (tunv) [inline app/views/users/new.html.erb]: change ' to "

File: app/views/users/index.html.erb [hunk 6]
```diff
@@ -0,0 +1,6 @@
+<h1><%= t ".h1index" %></h1>
+<%= will_paginate %>
+<ul class="users">
+  <%= render @users %>
+</ul>
+<%= will_paginate %>
```
Comment (tunv) [inline app/views/users/index.html.erb]: duplicate code

File: app/models/user.rb [hunk 7]
```diff
@@ -0,0 +1,10 @@
+class User < ActiveRecord::Base
+  before_save{self.email = email.downcase}
+  validates :name, presence: true
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+  validates :email, presence: true, length: {maximum: 255},
```
Comment (tunv) [inline app/models/user.rb]: add to rails config

File: .rubocop.yml [hunk 8]
```diff
@@ -0,0 +1,543 @@
...
+# Put the following files into the same directory as this file.
+inherit_from:
+  - .rubocop_enabled.yml
+  - .rubocop_disabled.yml
+
+# Common configuration.
+AllCops:
+  # Include gemspec and Rakefile
+  Include:
+    - '**/*.gemspec'
```
Comment (thanhnga67) [inline .rubocop.yml]: '=>"

File: app/helpers/users_helper.rb [hunk 9]
```diff
@@ -0,0 +1,8 @@
+module UsersHelper
+  # Returns the Gravatar for the given user.
+  def gravatar_for user
+    gravatar_id = Digest::MD5.hexdigest(user.email.downcase)
+    gravatar_url = "https://secure.gravatar.com/avatar/#{gravatar_id}?s=#{size}"
```
Comment (phuongbk) [inline app/helpers/users_helper.rb]: Cho vào Setting

File: app/views/shared/_error_messages.html.erb [hunk 10]
```diff
@@ -0,0 +1,12 @@
+<% if @user.errors.any? %>
+  <div id="error_explanation">
+    <div class="alert alert-danger">
+      The form contains <%= pluralize(@user.errors.count, "error") %>.
```
Comment (phuongbk) [inline app/views/shared/_error_messages.html.erb]: I18n

File: app/views/layouts/_shim.html.erb [hunk 11]
```diff
@@ -0,0 +1,4 @@
+<!--[if lt IE 9]>
+  <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/r29/html5.min.js">
+  </script>
+<![endif]-->
```
Comment (phuongbk) [inline app/views/layouts/_shim.html.erb]: dont'e need. Delete it

File: app/controllers/users_controller.rb [hunk 12]
```diff
@@ -1,4 +1,8 @@
 class UsersController < ApplicationController
+  def index
+    @users = User.paginate(page: params[:page], per_page: Settings.user.per_page)
```
Comment (yuyuvn) [inline app/controllers/users_controller.rb]: remove ()

File: app/views/users/_user.html.erb [hunk 13]
```diff
@@ -0,0 +1,4 @@
+<li>
+  <%= gravatar_for user%>
```
Comment (phuongbk) [inline app/views/users/_user.html.erb]: indent

**Pull request merged**